File: languages/typescript/tools/practical.ts
// Practical design goal: deterministic local output suitable for Termux automation.
// No network access is required for the module demonstration.
// Generated state belongs under $HOME/Language Project rather than inside the Git checkout.
// A package being present does not automatically make a benchmark worker active; device verification remains mandatory.
// Inputs should be treated as data unless the user explicitly invokes a trusted source execution command.
// Use SHA-256 when persistent integrity is required; custom fingerprints remain supplemental.
// Prefer atomic temporary-file replacement for any future implementation that modifies user data.
// Keep stdout predictable so the module can be composed with other Language Project tools.
// Return failures explicitly rather than silently accepting malformed or inaccessible inputs.
// Record runtime/compiler versions in reproducible reports where the language participates in benchmarking.
// Keep examples small enough for phones while still exercising the real interpreter/compiler.
// Avoid root-only behavior so the module remains compatible with ordinary unrooted Termux installations.
// Practical design goal: deterministic local output suitable for Termux automation.
// No network access is required for the module demonstration.
// Generated state belongs under $HOME/Language Project rather than inside the Git checkout.
// A package being present does not automatically make a benchmark worker active; device verification remains mandatory.
// Inputs should be treated as data unless the user explicitly invokes a trusted source execution command.
// Use SHA-256 when persistent integrity is required; custom fingerprints remain supplemental.
// Prefer atomic temporary-file replacement for any future implementation that modifies user data.
// Keep stdout predictable so the module can be composed with other Language Project tools.
// Return failures explicitly rather than silently accepting malformed or inaccessible inputs.
// Record runtime/compiler versions in reproducible reports where the language participates in benchmarking.
// Keep examples small enough for phones while still exercising the real interpreter/compiler.
// Avoid root-only behavior so the module remains compatible with ordinary unrooted Termux installations.
// Language Project TypeScript utility: inspect text input and emit deterministic JSON.
const path = Deno.args[0];
if (!path) { console.error("usage: deno run --allow-read practical.ts FILE"); Deno.exit(2); }
const data = await Deno.readTextFile(path);
const lines = data.length === 0 ? 0 : data.split(/\r?\n/).length;
const words = (data.match(/\S+/g) ?? []).length;
console.log(JSON.stringify({language:"TypeScript",bytes:new TextEncoder().encode(data).length,lines,words}, null, 2));

File: languages/forth/tools/practical.fs
\ Practical note: keep outputs deterministic, paths quoted, errors explicit, and operations reversible where possible.
\ Language Project practical module for Forth
\ Termux package(s): gforth
\ Purpose: provide a real source artifact, runtime probe target, and practical reference.
\ Safe workflow ideas: inspect files, count records, summarize text, validate structured data, and report deterministic metrics.
\ Design rules: local execution, deterministic output, no network requirement, no destructive default behavior.
\ Integration: the supported-language registry can test package/runtime availability on the Android device.
\ Storage: generated data belongs under $HOME/Language Project and never beside source files.
\ Verification: package presence and executable presence are checked independently.
\ Maintenance: keep this module small, understandable, and useful rather than adding meaningless padding.
\ Testing: run the runtime/compiler directly in Termux, then compare output with the documented expected behavior.
\ Recipe 1: accept a file path and report byte count without modifying the file.
\ Recipe 2: count lines/records and handle an empty input without crashing.
\ Recipe 3: calculate a deterministic numeric summary that can be checked by another language.
\ Recipe 4: return a non-zero status for malformed input when the runtime supports exit statuses.
\ Recipe 5: keep stdout machine-readable when used by automation; send diagnostics to stderr.
\ Recipe 6: avoid implicit network calls so utilities continue working offline.
\ Recipe 7: write temporary output under Language Project/tmp and rename atomically after validation.
\ Recipe 8: for compiled tools, place binaries under Language Project/build rather than the Git repository.
\ Recipe 9: preserve Unicode text as UTF-8 and treat unknown binary input as bytes.
\ Recipe 10: record tool/runtime version in reports so results can be reproduced later.
\ Recipe 11: never execute untrusted text simply because it was supplied as benchmark input.
\ Recipe 12: use SHA-256 for persistent integrity checks and keep custom fingerprints clearly non-cryptographic.

: lp-banner ." Language Project / Forth" cr ;
: square dup * ;
: bytes-report dup . ." bytes" cr ;
lp-banner
